src/linked_list.py

```diff
@@ -48,35 +48,38 @@ def size(self):
         return count
 
     def __len__(self):
-        """Function binds LinkedList to __len__ builtin so len() works."""
+        """Binds .size to __len__ builtin so len() works."""
         return self.size()
 
     def search(self, data):
-        """Function builds search method of LinkedList class."""
+        """Feturns the node that a specific value is in."""
         current = self.head
         try:
-            while current is not None and data != current.data:
+            while data != current.data:
                 current = current.next_node
             return current
-        except:
-            return None
+        except AttributeError:
+            raise IndexError("That value is not in the list.")
 
     def remove(self, node):
-        """Function build remove method of LinkedList class."""
+        """Removes a specific node from the list."""
         current = self.head
-        while current is not None and node.data != current.data:
-            previous = current
-            current = current.next_node
-        previous.next_node = current.next_node.next_node
-        return current
+        try:
+            while node.data != current.data:
```

Comment:
This falls victim to the same vulnerability that Cris pointed out earlier. What's the difference between:
```python
node != current
```
and 

```python
node.data != current.data
```